include/aws/common/atomics_msvc.inc

```diff
@@ -0,0 +1,368 @@
+ * memory models. We may need to make use of platform-specific intrinsics.
+ *
+ * NOTE: Before removing this #error, please make use of the Interlocked*[Acquire|Release]
+ * variants (if applicable for the new platform)! This will (hopefully) help ensure that
+ * code breaks before people take too much of a dependency on it.
+ */
+
+#endif
+
+/**
+ * Some general notes:
+ *
+ * On x86/x86_64, by default, windows uses acquire/release semantics for volatile accesses;
+ * however, this is not the case on ARM, and on x86/x86_64 it can be disabled using the
+ * /volatile:iso compile flag.
+ *
+ * Interlocked* functions implicitly have acq_rel semantics; there are ones with weaker
+ * semantics as well, but because windows is generally used on x86, where there's not a lot
+ * of performance difference between different ordering modes anyway, we just use the stronger
+ * forms for now. Further, on x86, they actually have seq_cst semantics as they use locked instructions.
+ * It is unclear if Interlocked functions guarantee seq_cst on non-x86 platforms.
+ *
+ * Since all loads and stores are acq and/or rel already, we can do non-seq_cst loads and stores
+ * as just volatile variable accesses, but add the appropriate barriers for good measure.
+ *
+ * For seq_cst accesses, we take advantage of the facts that (on x86):
+ * 1. Loads are not reordered with other loads
+ * 2. Stores are not reordered with other stores
+ * 3. Locked instructions (including swaps) have a total order
+ * 4. Non-locked accesses are not reordered with locked instructions
+ *
+ * Therefore, if we ensure that all seq_cst stores are locked, we can establish
+ * a total order on stores, and the intervening ordinary loads will not violate that total
+ * order.
+ * See http://www.cs.cmu.edu/~410-f10/doc/Intel_Reordering_318147.pdf 2.7, which covers
+ * this use case.
+ */
+
+#pragma warning(push)
+#pragma warning(disable : 4201) /* nameless union */
+
+#ifdef _M_IX86
+#    define AWS_INTERLOCKED_INT(x) Interlocked##x
+typedef int aws_atomic_impl_int_t;
+#else
+#    define AWS_INTERLOCKED_INT(x) Interlocked##x##64
+typedef int64_t aws_atomic_impl_int_t;
+#endif
+
+struct aws_atomic_var {
+    union {
+        aws_atomic_impl_int_t intval;
+        void *ptrval;
+    };
+};
+
+#pragma warning(pop)
+
+static inline void aws_atomic_priv_check_order(enum aws_memory_order order) {
+#ifndef NDEBUG
+    switch (order) {
+        case aws_memory_order_relaxed:
+            return;
+        case aws_memory_order_acquire:
+            return;
+        case aws_memory_order_release:
+            return;
+        case aws_memory_order_acq_rel:
+            return;
+        case aws_memory_order_seq_cst:
+            return;
+        default: /* Unknown memory order */
+            abort();
+    }
+#endif
+    (void)order;
+}
+
+enum aws_atomic_mode_priv { aws_atomic_priv_load, aws_atomic_priv_store, aws_atomic_priv_load_store };
```

Comment:
`aws_atomic_priv_load_store` isn't used anywhere

Reply:
I'll remove it.